### PR TITLE
Bump Rubocop to v0.58.x

### LINKED ISSUE #7115: Update dependency constraint to allow for rubocop v0.58.0
Hey there! :wave:

I noticed that the constraint you have for rubocop doesn't allow for the latest version to be used.

The constraint I found was `~> 0.57.2`, and the latest version available is `0.58.0`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### LINKED ISSUE #7156: Update dependency constraint to allow for rubocop v0.58.2
Hey there! :wave:

I noticed that the constraint you have for rubocop doesn't allow for the latest version to be used.

The constraint I found was `~> 0.57.2`, and the latest version available is `0.58.2`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### PR DESCRIPTION
**Pull request by ashmaroli**
Resolves #7156

### CODE REVIEW

**Comment by ashmaroli** [comment]
Opened a ticket at Rubocop's repo regarding the bug here: https://github.com/rubocop-hq/rubocop/issues/6084

**Comment by ashmaroli** [comment]
Note: This issue has not been resolved by `rubocop-0.58.1`

**Comment by kenman345** [comment]
@ashmaroli I am giving things a go at fixing the issue. https://github.com/rubocop-hq/rubocop/pull/6095

Its not a difficult change to make. It takes only a one or two line change its just knowing what the effective result should be and if any why/how it changed and following the proper PR steps from them. I am not as familiar with doing things the way they do it but I should be okay.

**Comment by ashmaroli** [comment]
@kenman345 Thank you very much for attempting to patch the bug. :+1: 
Hopefully, `rubocop-0.58.2` soon ships with the patch..

**Comment by kenman345** [comment]
Well the PR is approved so it should only be a matter of time before it’s merged and then released in the next cycle. 

They seem to only have a few PRs open at any time, I would imagine they are diligent about merging in approved requests and releasing builds when appropriate.

**Comment by DirtyF** [comment]
Shouldn't we rely on rubocop-jekyll gem here?

**Comment by kenman345** [comment]
https://github.com/rubocop-hq/rubocop/releases/tag/v0.58.2

@ashmaroli v0.58.2 of Rubocop has been cut and it has my fix. Please test/update this PR!

**Comment by ashmaroli** [comment]
@kenman345 updated already. Thanks for patching the issue that was blocking this :+1:

**Comment by kenman345** [comment]
No Problem! The solution wasnt bad but really its odd they're okay with the oddity in how it still works. I made sure that my PR included tests to show out the oddities though so its easy to explain and patch up in the future if they would like.

**Comment by ashmaroli** [comment]
Hope this gets merged before JekyllBot opens a ticket *recommending* us to relax constraints to allow using Rubocop-0.58.2 :stuck_out_tongue:

File: Gemfile [hunk 1]
```diff
@@ -25,7 +25,7 @@ group :test do
   gem "nokogiri", "~> 1.7"
   gem "rspec"
   gem "rspec-mocks"
-  gem "rubocop", "~> 0.57.2"
+  gem "rubocop", "~> 0.58.0"
```
Comment (pathawks) [inline Gemfile]: This should def be `0.58.2`, since anything less breaks stuff.

**Comment by pathawks** [comment]
I don't really understand what the plan here is with regards to **rubocop-jekyll**. I will defer to somebody who has been following that roadmap more carefully.

**Comment by ashmaroli** [comment]
@pathawks The reason `jekyll/jekyll` hasn't sswitched to `rubocop-jekyll` yet is to allow the Jekyll Repository to *"dictate"* the styles that needs to be followed with respect to the latest Rubocop version.

`rubocop-jekyll` simply mirrors the configuration (at `jekyll/jekyll:master`) via a separate gem so that the plugins that conform to "Jekyll Style" do not have to wait for a new version of `jekyll` gem to be released.

**Comment by kenman345** [comment]
PR looks good to me. Question below is about Jekyll-rubocop


@ashmaroli Should perhaps we move Jekylls rubocop methods to be instead the latest from that repository? We can make it into a [submodule ](https://blog.github.com/2016-02-01-working-with-submodules/) on this repository pointing to that repository, so that we only make updates and changes in one place for those.

**Comment by kenman345** [comment]
So is this not being merged? Its passing and the requested changes were made. The discussion can continue without halting this PR. Would be nice to be able to use the latest from RuboCop

**Pull request closed without merging**